packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -16,6 +19,47 @@ class AndroidCameraCameraX extends CameraPlatform {
   /// Returns list of all available cameras and their descriptions.
   @override
   Future<List<CameraDescription>> availableCameras() async {
-    throw UnimplementedError('availableCameras() is not implemented.');
+    final List<CameraDescription> cameraDescriptions = <CameraDescription>[];
+
+    ProcessCameraProvider processCameraProvider =
+        await ProcessCameraProvider.getInstance();
+    final List<CameraInfo> cameraInfos =
+        await processCameraProvider!.getAvailableCameraInfos();
+
+    final CameraSelector backCameraSelector =
+        CameraSelector.getDefaultBackCamera();
+    final CameraSelector frontCameraSelector =
+        CameraSelector.getDefaultFrontCamera();
+
+    CameraLensDirection cameraLensDirection;
+    int cameraCount = 0;
+    int cameraSensorOrientation;
+    String cameraName;
+
+    for (final CameraInfo cameraInfo in cameraInfos) {
+      // Determine the lens direction by filtering the CameraInfo
+      // TODO(gmackall): replace this with call to CameraInfo.getLensFacing when changes containing that method are available
+      if ((await backCameraSelector.filter(<CameraInfo>[cameraInfo]))
+          .isNotEmpty) {
+        cameraLensDirection = CameraLensDirection.back;
+      } else if ((await frontCameraSelector.filter(<CameraInfo>[cameraInfo]))
+          .isNotEmpty) {
+        cameraLensDirection = CameraLensDirection.front;
+      } else {
+        //Skip this CameraInfo as its lens direction is unknown
+        continue;
+      }
+
+      cameraSensorOrientation = await cameraInfo.getSensorRotationDegrees();
+      cameraName = 'Camera $cameraCount';
```

Comment:
Is this how Android named cameras with the method used in the old plugin?https://developer.android.com/reference/android/hardware/camera2/CameraManager#getCameraIdList()?